mergin/client.py

```diff
@@ -322,6 +324,43 @@ def user_service(self):
 
         return response
 
+    def server_type(self):
+        """
+        Returns the deployment type of the server:
+        ee - server supports workspaces
+        ce - server has a global workspace
+        old - server does not support workspaces
+
+        The value is cached for self's lifetime
+
+        :returns: Type of server deployment ("ee"/"ce"/"old")
+        rtype: str
+        """
+        # todo: return enum
+        if not self._server_type:
+            try:
+                resp = self.get("/config")
+                config = json.load(resp)
+                if "user_workspaces_allowed" in config:
+                    self._server_type = "ee"
+                if "global_namespace" in config:
+                    self._server_type = "ce"
+                    self._global_namespace = config["global_namespace"]
+            except ClientError as e:
+                self._server_type = "old"
```

Comment:
Just a note for the future: `/config` will return the type of the server explicitly without the need to guess it this way.